### PR TITLE
multi: remove mock.go

### PR DESCRIPTION
**Pull request by Crypt-iQ**
This PR removes `mock.go` and allows us to import these mocks when we couldn't before. This will greatly aid in testing.

### CODE REVIEW

**Comment by Crypt-iQ** [comment]
`travis-race` is a flake, itest is also a flake. Should be good to go.

File: lnwallet/mock.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
+package lnwallet
```
Comment (halseth) [inline lnwallet/mock.go:1]: ideally mocks should be in files ending with `_test.go` to avoid compiler checking them during non-testing.

**Review by halseth — COMMENTED** [review]
Can you add a short rationale to the PR description?

**Comment by Crypt-iQ** [comment]
Unit tests pass on my machine and I didn't modify the integration tests. The linter complains about `mock.MockChainIO` being too stutter-y. Not sure if I should drop the `Mock` prefix for these structs?

Additionally, I'm not sure how to structure the `lntest/mock` directory. Currently, all the mock files are in the `lntest/mock` directory and I'm wondering if we can do better. Perhaps something like `lntest/mock/lnwallet/` where this folder contains all the mocks for things we want to mock in the `lnwallet` directory? Or even a new top-level `mock` package altogether? 

@halseth @cfromknecht

**Comment by halseth** [comment]
I think everything in `lntest/mock` is good enough for now.

For linting, sounds good to use naming convention `mock.<whatever-we-are-mocking>`

**Comment by Crypt-iQ** [comment]
Travis failures should be flakes. Should be ready for review now that linting passes.

**Review by halseth — APPROVED** [review]
Nice change, LGTM 🧵

**Review by cfromknecht — APPROVED** [review]
LGTM. Great commit structure 🎉

**Pull request merged**